src/Mvc/ResourceResponseListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrRest\Mvc;
+
+use Zend\EventManager\AbstractListenerAggregate;
+use Zend\EventManager\EventManagerInterface;
+use Zend\Http\Response as HttpResponse;
+use Zend\Mvc\MvcEvent;
+
+/**
+ * HttpExceptionListener
+ *
+ * @license MIT
+ * @author  Michaël Gallego <mic.gallego@gmail.com>
+ */
+class ResourceResponseListener extends AbstractListenerAggregate
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function attach(EventManagerInterface $events)
+    {
+        $this->listeners[] = $events->attach(MvcEvent::EVENT_FINISH, [$this, 'finishResponse'], -500);
+    }
+
+    /**
+     * Get the exception and optionally set status code, reason message and additional errors
+     *
+     * @internal
+     * @param  MvcEvent $event
+     * @return void
+     */
+    public function finishResponse(MvcEvent $event)
+    {
+        $response = $event->getResponse();
+
+        if (!$response instanceof HttpResponse) {
+            return;
+        }
+
+        $method = strtolower($event->getRequest()->getMethod());
+
+        switch ($method) {
+            case 'delete':
+                // According to http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.7, status code should
+                // be 204 if nothing is returned
+                if (empty($response->getContent())) {
+                    $response->setStatusCode(204);
```

Comment:
Yeah, I'm not satisfied with it either for now. But we don't have links and so on, so we cannot do much yet. For now I just wanted to do it as simple as possible, but yeah, this will need to be fixed in the future.